presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -106,9 +106,13 @@ public NativeExecutionTask(
                 this.workerClient,
                 this.executor,
                 taskManagerConfig.getInfoUpdateInterval());
-        this.taskResultFetcher = new HttpNativeExecutionTaskResultFetcher(
-                updateScheduledExecutor,
-                this.workerClient);
+        if (!shuffleWriteInfo.isPresent()) {
+            this.taskResultFetcher = Optional.of(new HttpNativeExecutionTaskResultFetcher(
+                    updateScheduledExecutor,
+                    this.workerClient));
+        } else {
+            this.taskResultFetcher = Optional.empty();
+        }
```

Comment:
Maybe can be a one liner
```
this.taskResultFetcher = !shuffleWriteInfo.isPresent() ? new HttpNativeExecutionTaskResultFetcher(
                    updateScheduledExecutor,
                    this.workerClient) : Optional.empty();
```

Reply:
I find this harder to read and prefer to keep it the way it is now.